src/Akka.HealthCheck/Transports/LivenessTransportActor.cs

```diff
@@ -19,15 +22,15 @@ namespace Akka.HealthCheck.Transports
     public sealed class LivenessTransportActor : ReceiveActor
     {
         private const int LivenessTimeout = 1000;
-        private readonly IActorRef _livenessProbe;
+        private readonly List<IActorRef> _livenessProbes;
         private readonly ILoggingAdapter _log = Context.GetLogger();
         private readonly IStatusTransport _statusTransport;
         private readonly bool _logInfo;
 
-        public LivenessTransportActor(IStatusTransport statusTransport, IActorRef livenessProbe, bool log)
+        public LivenessTransportActor(IStatusTransport statusTransport, ImmutableDictionary<string, IActorRef> livenessProbes, bool log)
         {
             _statusTransport = statusTransport;
-            _livenessProbe = livenessProbe;
+            _livenessProbes = livenessProbes.Values.ToList();
             _logInfo = log;
 
             ReceiveAsync<LivenessStatus>(async status =>
```

Comment:
So one question - if we have multiple healthchecks:

- Cluster
- Remote
- Persistence

Suppose persistence fails but cluster and remoting are still sending happy thoughts back to the `LivenessTransportActor` - do the happy thoughts from the two live probes override the "things are bad" signal from the probe that is failing? I.e. do we create a false negative here when that happens?

If so, shouldn't it be that we need some more robust boolean logic here:

1. If _any_ of the probes are in a failing state, the entire liveness check is failed
2. If _none_ of the probes are in a failing state, we're live

Reply:
Got it, I'll implement this.